api/src/main/java/org/openmrs/module/fhir/api/impl/PersonServiceImpl.java

```diff
@@ -114,4 +116,23 @@ public Person updateFHIRPerson(Person thePerson, String theId) {
 			return createFHIRPerson(thePerson);
 		}
 	}
+
+	/**
+	 * @see org.openmrs.module.fhir.api.PersonService#retirePerson(String)
+	 */
+	@Override
+	public void retirePerson(String id) throws ResourceNotFoundException, NotModifiedException {
+		org.openmrs.Person person = Context.getPersonService().getPersonByUuid(id);
+		if (person == null) throw new ResourceNotFoundException(String.format("Person with id '%s' not found", id));
+		if (person.isPersonVoided()) return;
+		try {
+			person.setVoided(true);
```

Comment:
SInce AOP using within Spring, some handlers are triggered when invoking the apis. Can you have a quick look at database and see whether it's setting those values?